File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    # 누적합 배열의 두 원소 차가 n이 되는 (i, j) 쌍의 개수
    arr = [0]
    for i in range(1,n+1):
        arr.append(arr[-1] + i)
    l = 0
    r = 0
    ans = 0
    while r < len(arr):
        s = arr[r] - arr[l]
        if s == n:
            ans += 1
            r += 1
        elif s > n:
            l += 1
        else: r += 1
    return ans

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
import java.util.*;
class Solution {
    public int solution(int n) {
        List<Integer> arr = new ArrayList<>();
        arr.add(0);
        int answer = 0;
        for (int i = 1; i <= n; i++) {
            arr.add(arr.get(arr.size() - 1) + i);
        }
        int l = 0;
        int r = 0;
        while (r < arr.size()) {
            int sum = arr.get(r) - arr.get(l);
            if (sum == n) {
                answer++;
                r++;
            }
            else if (sum > n) {
                l++;
            } else {
                r++;
            }
        }
        return answer;
    }
}